packages/mui/src/FieldTemplate/FieldTemplate.tsx

```diff
@@ -25,11 +23,18 @@ const FieldTemplate = ({
   uiSchema,
   registry,
 }: FieldTemplateProps) => {
+  const uiOptions = getUiOptions(uiSchema);
+  const WrapIfAdditionalTemplate = getTemplate(
```

Comment:
Adding the string name as the generic will ensure the proper typing
```suggestion
  const WrapIfAdditionalTemplate = getTemplate<"WrapIfAdditionalTemplate">(
```